cubi_tk/snappy/snappy_workflows.py

```diff
@@ -0,0 +1,156 @@
+
+        config, lookup_paths, config_paths = self._expand_ref(
+            str(config_path),
+            config_data,
+            [
+                str(workflow_step_path),
+                str(pathlib.Path(workflow_step_path).parent / ".snappy_pipeline"),
+            ],
+        )
+        return config, lookup_paths, config_paths
+
+    def _get_workflow_name(
+        self, workflow_path: typing.Union[str, pathlib.Path]
+    ) -> typing.Optional[str]:
+        """Get the name of the workflow in the directory. This will parse the contained config.yaml."""
+
+        config, _, _ = self._load_workflow_step_configuration(workflow_path)
+        if config is not None and "pipeline_step" in config:
+            return config["pipeline_step"].get("name", None)
+
+    def get_snappy_step_directories(
+        self, snappy_root_dir: typing.Union[str, pathlib.Path]
+    ) -> typing.Dict[str, pathlib.Path]:
+        """Get a dictionary of snappy workflow step names and their associated path.
+
+        :param snappy_root_dir: Path to the snappy root directory, also containing .snappy_pipeline
+
+        :return: Dict of workflow step name to workflow step path.
+        """
+        folder_steps = {
+            name: path
+            for name, path in [
+                (self._get_workflow_name(p), p) for p in pathlib.Path(snappy_root_dir).iterdir()
+            ]
+            if name in self._step_to_module
+        }
+
+        return folder_steps
+
+    def get_workflow_step_dependencies(
+        self, workflow_step_path: typing.Union[str, pathlib.Path]
+    ) -> typing.List[str]:
+        """Find workflow dependencies for the given workflow step.
+        :param workflow_step_path: Path to the workflow step.
+
+        :return: List of dependencies for the given workflow step.
+        """
+        workflow_step_path = pathlib.Path(workflow_step_path)
+
+        config, lookup_paths, config_paths = self._load_workflow_step_configuration(
+            workflow_step_path
+        )
+        if config is None:
+            raise RuntimeError(
+                f"Could not load workflow step confiuration for {workflow_step_path}"
+            )
+
+        step_name = config["pipeline_step"]["name"]
+
+        module_snakefile = self._snakefile_path(step_name)
+
+        # get the name of the workflow step class name
+        obj_name = get_workflow_snakefile_object_name(module_snakefile)
+        if obj_name is None:
+            raise RuntimeError(f"Could not find workflow object for {step_name}")
+
+        workflow_module = importlib.import_module(f".{step_name}", "snappy_pipeline.workflows")
+        workflow_class = getattr(workflow_module, obj_name)
+        assert workflow_class.name == step_name
+
+        workflow_object = workflow_class(
+            DummyWorkflow(), config, lookup_paths, config_paths, str(workflow_step_path)
+        )
+        dependencies = workflow_object.sub_workflows.keys()
```

Comment:
I'm unsure if this is how snappy 'intends' to define dependencies, since there also should be `workflow_object.previous_steps`. Likely is either is fine though.